src/Mvc/Mvc.Razor.RuntimeCompilation/src/AssemblyPartExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationParts
+{
+    public static class AssemblyPartExtensions
+    {
+        /// <inheritdoc />
+        public static IEnumerable<string> GetReferencePaths(this AssemblyPart assemblyPart)
+        {
+            var assembly = assemblyPart?.Assembly ?? throw new ArgumentNullException(nameof(assemblyPart));
```

Comment:
This should do a cast to the interface. Was the interface removed?

Reply:
IMO it's better if the interface is a detail of this extension method.

Reply:
I don't follow. What does this look like?

Reply:
Nevermind my brain is crazy.